community/kernel/src/test/java/org/neo4j/kernel/impl/proc/ProcedureJarLoaderTest.java

```diff
@@ -199,6 +204,40 @@ public void shouldGiveHelpfulErrorOnGenericStreamProcedure() throws Throwable
         jarloader.loadProcedures( jar );
     }
 
+    @Test
+    public void shouldGiveHelpfulErrorOnUnsafeRestrictedProcedure() throws Throwable
+    {
+        // Given
+        URL jar = createJarFor( ClassWithUnsafeComponent.class );
+
+        // Expect
+        exception.expect( ProcedureException.class );
+        exception.expectMessage( "Unable to set up injection for procedure `ClassWithUnsafeComponent`, the field" +
+                " `api` has type `class org.neo4j.kernel.impl.proc.ProcedureJarLoaderTest$UnsafeAPI` which is not a" +
+                " known injectable component.");
```

Comment:
Could there be different error messages for components that actually exists but are unsafe, and those that do not exists at all? That would be even more helpful.